src/xconnection.cpp

```diff
@@ -389,22 +385,63 @@ std::experimental::optional<vector<Window>>
 std::experimental::optional<vector<string>>
     XConnection::getWindowPropertyTextList(Window window, Atom property)
 {
-    XTextProperty text_prop;
-    if (!XGetTextProperty(m_display, window, &text_prop, property)) {
+    Atom prop_type;
+    int format;
+    unsigned long bytes_left;
+    char* items_return; // TODO: char* or unsigned char*?
+    unsigned long count;
+    int status = XGetWindowProperty(m_display, window,
+            property, 0, ULONG_MAX, False, AnyPropertyType,
+            &prop_type, &format, &count, &bytes_left,
+            (unsigned char**)&items_return);
+    if (Success != status || prop_type == None || format == 0) {
         return {};
     }
-    char** list_return;
-    int count;
-    if (Success != Xutf8TextPropertyToTextList(m_display, &text_prop, &list_return, &count)) {
-        XFree(text_prop.value);
-        return {};
+    if (format != 8) {
+        fprintf(stderr, "herbstluftwm error: can not parse the "
+                        " atom \'%s\' of window 0x%lx: expected format=8 but got"
+                        " format=%d\n",
+                        XGetAtomName(m_display, property), window, format);
+        XFree(items_return);
     }
+    unsigned long offset = 0;
     vector<string> arguments;
-    for (int i = 0; i < count; i++) {
-        arguments.push_back(list_return[i]);
+    // the trailing 0 at items_return[count] might be crucial:
+    // if the string list ends with the empty string, then we
+    // need to access items_return[count].
+    while (offset <= count) {
+        char* textChunk = items_return + offset;
+        // let us hope that items_return is properly null-byte terminated.
```

Comment:
This sounds unsettling :D

Reply:
I think I once read in the docs that `items_return` always returns one element more than `count` and that it's always null, but I couldn't find it anymore when writing this PR.

Reply:
Fingers crossed...